kubernetes/provider.go

```diff
@@ -234,7 +233,43 @@ func (k kubeClientsets) AggregatorClientset() (*aggregator.Clientset, error) {
 	return k.aggregatorClientset, nil
 }
 
+func inCluster() bool {
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	return host != "" && port != ""
+}
+
+func checkConfigurationValid(d *schema.ResourceData) error {
```

Comment:
@dak1n1 This should generate a more helpful error if the user hasn't supplied any config through the provider block or environment variables, or terraform isn't inside a cluster. I've tried a few incantations locally but would appreciate a second 👀 .

```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


------------------------------------------------------------------------

Error: provider not configured: you must configure a path to your kubeconfig
or explicitly supply credentials in the provider block
```

Reply:
Nice! This is exactly what I was hoping for. As a user, if I got this message, the next thing I would do is open up the documentation for the provider, to figure out how to specify this. (Or google the error message because I don't want to navigate through all the docs pages).

But we might be able to save them that step by giving them either the name of the environment variable, or the name of the attribute to put in their config. Something like `KUBE_CONFIG_PATHS=$KUBECONFIG` should work.

I also noticed that when `KUBE_CONFIG_PATH` and `KUBE_CONFIG_PATHS` are both set, it uses `KUBE_CONFIG_PATH`. And if `KUBE_CONFIG_PATH` points to a bad path, it gives this error:

```
$ env |grep KUBE
KUBE_CONFIG_PATH=/bad/path
KUBECONFIG=/home/dakini/.kube/config
KUBE_CONFIG_PATHS=/home/dakini/.kube/config

$ terraform apply --auto-approve
kubernetes_pod.test: Creating...

Error: Post "http://localhost/api/v1/namespaces/default/pods": dial tcp [::1]:80: connect: connection refused
```

It might be worth checking that the file exists, so the error message tells the user what went wrong.

Reply:
We could put a hint, but we can't assume which method they actually need - and writing instructions for path, basic auth, TLS creds, env vars is going to be quite a lot of information to put in the error message. Perhaps we could just link them to the documentation page for the provider block?

Reply:
Also yes, confirming those files even exist early on would be a good idea – I'll add that to where it does the path expansion.